extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -110,10 +111,10 @@ abstract class ControllerDialogBase extends InitializingComponent {
 	public abstract async validate(): Promise<boolean>;
 
 	private handleCancel(): void {
-		this.completionPromise.resolve(undefined);
+		this.completionPromise.reject(new UserCancelledError(loc.userCancelledError));
```

Comment:
I guess this is fine - but is there any particular reason that you chose to throw an error instead of just continuing to return undefined? A couple thoughts about that : 

1. The VS Code APIs tend to just returned undefined when the dialog/popup is cancelled (such as quick pick items) so doing something different here is a bit confusing
1. A user cancelling generally isn't really an error so doesn't seem like something we should have throw
1. We only returned undefined if it's cancelled so it's not like there's other cases we're needing to take care of

Reply:
I was not aware of vscode apis returning undefined for cancel. In general, do not like the use of sentinel values for communicating state. I think the treatment of 'cancellation' as success or failure is where the difference in approach is.

I will revert to the old behavior.

Reply:
reveted